alf/algorithms/algorithm.py

```diff
@@ -1020,7 +1020,12 @@ def update_with_gradient(self,
 
         if isinstance(loss_info.loss, torch.Tensor):
             loss = weight * loss_info.loss
-            loss.backward()
+            # In theory, there might be unfinished kernel before backward is called
+            # and and after backward() returns and torch.cuda.synchronize() should
+            # be called to make sure we only measure the computation time for backward.
+            # In practice, we found that the measured time is almost unchanged.
```

Comment:
Not sure what this paragraph means?

Reply:
In general, there is no guarantee that its computation is finished when a torch function returns. See https://pytorch.org/docs/stable/notes/cuda.html#asynchronous-execution

Reply:
OK, I think it's a little difficult to parse the first sentence. Maybe 

"In theory, there might be unfinished kernel between backward being called and backward returning, and torch.cuda.synchronize() should be called to make sure..."